### PR TITLE
Viewer dataset

### PR DESCRIPTION
**Pull request by will-moore**
# What this PR does

When you open the image viewer for an image in a Dataset, we add ```?dataset=id``` to the URL so that the viewer knows which Dataset the image is in.

Although we don't use this info in the current viewer, if you configure the default viewer to be iViewer

```$ bin/omero config set omero.web.viewer.view omero_iviewer.views.index```

this will use the Dataset ID to load thumbnails:

screen shot 2017-03-28 at 16 35 32

Have also discussed pathviewer using this - cc @chris-allan.

# Testing this PR

1. Open a viewer by:
    - Double-clicking in tree
    - Double-clicking thumbnail in Dataset
    - Right-click on thumbnail in Dataset -> Open Link in New Tab
    - Click "Full Viewer" button in General tab
2. Viewer should open as normal, with URL ending with ```?dataset={id}```
3. Images in other containers (Tags, Orphaned, Shares) should render thumbnails OK and work as before.

### CODE REVIEW

**Comment by jburel** [comment]
@will-moore see https://github.com/openmicroscopy/management_tools/pull/354

**Comment by waxenegger** [comment]
Very handy feature. Checked on web-dev-merge and the dataset id is appended: 

open_dataset

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-DEV-merge-push#647](https://ci.openmicroscopy.org/job/OMERO-DEV-merge-push/647/). See the [console output](https://ci.openmicroscopy.org/job/OMERO-DEV-merge-push/647/consoleText) for more details.
Possible conflicts:
  - PR #5158 aleksandra-tarkowska 'Get thumbnails (rebased onto develop)'
    - components/tools/OmeroWeb/omeroweb/webclient/templates/webclient/data/icon_thumbnails_underscore.html
  - PR #5192 will-moore 'Get thumbnails'
    - components/tools/OmeroWeb/omeroweb/webclient/templates/webclient/data/icon_thumbnails_underscore.html

**Comment by pwalczysko** [comment]
Travis failure, but this PR seems to be in the merge, so I will test it.

**Comment by pwalczysko** [comment]
Works as expected on eel merge (web-dev-merge).

**Comment by pwalczysko** [comment]
Ready to merge FMPOV after the travis failure has been dealt with.

**Comment by will-moore** [comment]
Need to wait on Robot tests tomorrow to check fix of https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-robotframework/628/robot/Web%20&%20Web/Web_1/View%20Image/Test%20Open%20Viewer/

**Comment by jburel** [comment]
robot test still failing

**Comment by will-moore** [comment]
That test passed today: https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-robotframework/629/robot/Web%20&%20Web/Web_1/View%20Image/Test%20Open%20Viewer/

**Comment by pwalczysko** [comment]
screen shot 2017-06-16 at 14 51 04

The ``TestOpenViewer`` seems to not be in, but these tests failed. 
Not sure whether the failures are connected to this PR.

**Comment by will-moore** [comment]
@pwalczysko No, they're not due to this PR.

**Comment by pwalczysko** [comment]
happy to merge then

File: components/tools/OmeroWeb/omeroweb/webclient/static/webclient/javascript/ome.tree.js [hunk 1]
```diff
@@ -254,7 +254,16 @@ $(function() {
         if (node) {
             if (node.type === 'image') {
                 //Open the image viewer for this image
-                window.open(WEBCLIENT.URLS.webindex + "img_detail/" + node.data.obj.id, '_blank');
+                var url = WEBCLIENT.URLS.webindex + "img_detail/" + node.data.obj.id + "/";
+                // Add dataset id so the viewer can know it's context
```
Comment (jburel) [inline components/tools/OmeroWeb/omeroweb/webclient/static/webclient/javascript/ome.tree.js]: typo ``its``

**Comment by jburel** [comment]
Travis is green
Thanks

**Pull request merged**